cloud/common.go

```diff
@@ -0,0 +1,303 @@
+// Copyright 2024 by Red Hat, Inc. All rights reserved.
+// Use of this source is goverend by the Apache License
+// that can be found in the LICENSE file.
+
+package cloud
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"os"
+	"os/user"
+	"path/filepath"
+	"runtime"
+	"sort"
+	"strings"
+	"syscall"
+)
+
+// HTTPClient make it easier to swap out the client socket for testing
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+// NewClient initializes the values of the cloud API client configuration
+// used to query the server.
+func NewClient(ctx context.Context, socket HTTPClient, socketPath string) Client {
+	// TODO
+	// - check for valid server path
+	return Client{
+		ctx:        ctx,
+		socket:     socket,
+		socketPath: socketPath,
+		protocol:   "http",
+		host:       "localhost",
+		rawFunc:    func(string, string, int, []byte) {},
+	}
+}
+
+// InitClientUnixSocket configures the client to use a unix domain socket
+// This configures the cloud.Client with the socket path
+// It must be called before using any of the cloud.Client functions.
+func InitClientUnixSocket(ctx context.Context, socketPath string) Client {
```

Comment:
Since this returns a new client, I'd call it `NewClientFromUnixSocket` or such but that's minor.

Reply:
That's the same name used in the weldr client, so I'll leave it.